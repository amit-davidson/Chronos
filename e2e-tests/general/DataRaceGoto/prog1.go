package main

import (
	"sync"
)

var a int
var mutex1 = sync.Mutex{}

func main() {
	go func() {
		a = 5
	}()
	fn1()
}

func fn1() {
	if a > 4 {
		goto End
	}
	return
End:
	a = 6
}
