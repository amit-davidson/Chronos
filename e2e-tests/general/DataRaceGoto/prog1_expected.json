{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:12:3",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 1,
					"2": 1
				}
			}
		},
		{
			"ID": 2,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:12:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:7:5",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 1,
					"2": 2
				}
			}
		},
		{
			"ID": 3,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:18:5",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:7:5",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				}
			}
		},
		{
			"ID": 4,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:18:7",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:18:5",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 3
				}
			}
		},
		{
			"ID": 5,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:18:7",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 4
				}
			}
		},
		{
			"ID": 6,
			"Pos": "",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:18:7",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 5
				}
			}
		},
		{
			"ID": 7,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:23:2",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 6
				}
			}
		},
		{
			"ID": 8,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:23:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceGoto/prog1.go:7:5",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 7
				}
			}
		}
	]
}